### DISCUSSION TITLE
About vedantmgoyal2009/winget-pkgs-automation

### DISCUSSION THREAD

**Discussion by vedantmgoyal9** [microsoft/winget-pkgs#35516] [General] [answered]
I've created this discussion to keep things in one place rather than discussing them at the bottom of random PRs since it makes it hard to remember and find them.

@denelon We've been discussing the implementation of my automation into the winget-pkgs repository and you had already replied here: https://github.com/vedantmgoyal2009/winget-pkgs-automation/pull/144#issuecomment-974210334.

I was asking for the installation of the [GitHub App](https://github.com/apps/winget-pkgs-automation-bot) because of the following reasons:
- GitHub API limit is much more for GitHub Apps: https://docs.github.com/en/developers/apps/building-github-apps/rate-limits-for-github-apps#server-to-server-requests.
- It would allow me to approve PRs since PR authors can't approve their PRs. <img src="https://user-images.githubusercontent.com/83997633/142750169-57d6a873-22fb-4bda-8e90-fb25a86b8c19.png" width=60% height=60%>
This way I can help moderators to ease their work since I can approve PRs which I had validated myself manually.
- It would help in differentiating automatic vs. manual submissions.

I don't think any of the above things have an impact on the .package file or whatever method you choose for leveraging my work on the repo.

**Another Question:** What would you do when the "developer" himself wants to add the package in my project rather he publishing updates himself?

**Comment by denelon** [comment, marked as answer]
Re: Another Question:
We're trying to meet developers where they are. If for some reason they cannot automate publishing manifests as a part of their CI/CD pipeline, we certainly wouldn't deny them other methods of automation.
